test/middleware/Counter.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {BaseHook} from "./../../contracts/BaseHook.sol";
+import {Hooks} from "@uniswap/v4-core/src/libraries/Hooks.sol";
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {BeforeSwapDelta, BeforeSwapDeltaLibrary} from "@uniswap/v4-core/src/types/BeforeSwapDelta.sol";
+import {BalanceDelta, BalanceDeltaLibrary} from "@uniswap/v4-core/src/types/BalanceDelta.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/src/types/PoolId.sol";
+
+contract Counter is BaseHook {
+    using PoolIdLibrary for PoolKey;
+
+    mapping(PoolId => uint256) public beforeInitializeCount;
+    mapping(PoolId => uint256) public afterInitializeCount;
+
+    mapping(PoolId => uint256) public beforeSwapCount;
+    mapping(PoolId => uint256) public afterSwapCount;
+
+    mapping(PoolId => uint256) public beforeAddLiquidityCount;
+    mapping(PoolId => uint256) public afterAddLiquidityCount;
+    mapping(PoolId => uint256) public beforeRemoveLiquidityCount;
+    mapping(PoolId => uint256) public afterRemoveLiquidityCount;
+
+    mapping(PoolId => uint256) public beforeDonateCount;
+    mapping(PoolId => uint256) public afterDonateCount;
+
+    bytes public lastHookData;
```

Comment:
What's the use of this? And does it make sense to make it a mapping from selectors to lastHookData instead

Reply:
it's just for testing to make sure hookData is passed through correctly